app/models/manageiq/providers/embedded_automation_manager/configuration_script_source.rb

```diff
@@ -1,7 +1,115 @@
-class ManageIQ::Providers::EmbeddedAutomationManager::ConfigurationScriptSource <
-  ManageIQ::Providers::AutomationManager::ConfigurationScriptSource
+class ManageIQ::Providers::EmbeddedAutomationManager::ConfigurationScriptSource < ManageIQ::Providers::AutomationManager::ConfigurationScriptSource
+  virtual_attribute :verify_ssl, :integer
+
+  validates :name,       :presence => true # TODO: unique within region?
+  validates :scm_type,   :presence => true, :inclusion => { :in => %w[git] }
+  validates :scm_branch, :presence => true
+
+  supports :create
+
+  default_value_for :scm_type,   "git"
+  default_value_for :scm_branch, "master"
+
+  belongs_to :git_repository, :autosave => true, :dependent => :destroy
+  before_validation :sync_git_repository
 
   def self.display_name(number = 1)
     n_('Repository', 'Repositories', number)
   end
+
+  def git_repository
+    (super || (ensure_git_repository && super)).tap { |r| sync_git_repository(r) }
+  end
+
+  def verify_ssl=(val)
+    @verify_ssl = case val
+                  when 0, false then OpenSSL::SSL::VERIFY_NONE
+                  when 1, true  then OpenSSL::SSL::VERIFY_PEER
+                  else
+                    OpenSSL::SSL::VERIFY_NONE
+                  end
+
+    if git_repository_id && git_repository.verify_ssl != @verify_ssl
+      @verify_ssl_changed = true
+    end
+  end
+
+  def verify_ssl
+    if @verify_ssl
+      @verify_ssl
+    elsif git_repository_id
+      git_repository.verify_ssl
+    else
+      @verify_ssl ||= OpenSSL::SSL::VERIFY_NONE
+    end
+  end
+
+  def sync
+    update!(:status => "running")
+    transaction do
+      current = configuration_script_payloads.index_by(&:name)
+
+      git_repository.update_repo
+      git_repository.with_worktree do |worktree|
+        worktree.ref = scm_branch
+
+        yield current, worktree
```

Comment:
TODO I'd prefer to yield the file content since a.) it is used to confirm it is a valid playbook and b.) we want to store the content in the record similar to workflows.

This will require adding content to the configuration_scripts table, and also splitting up the valid file check so we're not reading every single file.

Reply:
I'm not sure - there may be cases where multiple files make up a single "thing" or multiple files are need to verify such "thing", so the subclass may want to walk the worktree itself and determine that for itself. I agree that we can only store the content for a single thing in a single column, but I kind of like the way it is in this PR where it gives the whole worktree.

Reply:
I do find it weird that the caller needs to delete the existing ones from `current`, and then the superclass method will takes what's left and destroy those - that part of the interface is weird to me, especially since if the caller forgets to remove it from `current`, then everything gets deleted.

Might be nicer for the caller's block to not touch `current`, but instead return what was "found".  Then the superclass method can remove the found from current and delete the rest.

Reply:
> Might be nicer for the caller's block to not touch current, but instead return what was "found". Then the superclass method can remove the found from current and delete the rest.

Yeah I was trying to do exactly this, ran into some trouble because the subclass uses `found` in `current` to update.  So 100% agree and am trying to get that working

Reply:
Okay I updated the child class to `find_or_create_by` and return the result, so the parent class can use the block results and compare to the old list

Reply:
hmmm- that might be worse unfortunately, because now it introduces N+1 queries.  I was more thinking it could pass in the current, so the query can be in-memory, but not modify that incoming current, instead returning the "new" current (Using .new for new records).